lms/services/launch_verifier.py

```diff
@@ -50,44 +49,78 @@ def verify(self):
             raise self._exception
 
     def _verify(self):
-        try:
-            consumer_key = self._request.params["oauth_consumer_key"]
-        except KeyError as err:
-            raise NoConsumerKey() from err
+        is_valid, _request = self._oauth1_endpoint.validate_request(
+            uri=self._request.url,
+            http_method=self._request.method,
+            body=self._request.body,
+            headers=dict(self._request.headers),
+        )
+
+        if not is_valid:
+            raise LTIOAuthError()
+
+
+class OAuthRequestValidator(RequestValidator):
+    # Value lifted from `oauth2` which is an `oauth1` library as was used by
+    # PyLTI. The default in `oauthlib` is 600
+    timestamp_lifetime = 300  # In seconds, five minutes.
+
+    # Tell oauthlib we are chill about http for local testing
+    enforce_ssl = False
+
+    def __init__(self, db_session):
+        super().__init__()
+        self.db_session = db_session
+
+    def check_client_key(self, client_key):
+        """Check that the client key only contains safe characters."""
+
+        # Just a style check for chars and length etc... we don't care about
+        # the format, as long as our lookup later works
+        return True
```

Comment:
It's mandatory to implement lots of methods on this, or the process crashes, but I don't actually think we want to enforce much about the style of the key.

`oauthlib`s design is very weird to me as well, as we implement all these different cases, but ultimately it just crushes them all down to `False` if anything fails. I'm not sure what the advantage in distinguishing is.